client/blocks/upsell-nudge/index.jsx

```diff
@@ -56,11 +56,14 @@ export const UpsellNudge = ( {
 		return null;
 	}
 
-	if ( ! href && site ) {
-		href = addQueryArgs( { feature, plan }, `/plans/${ site.slug }` );
-	}
+	const customerType = plan === PLAN_BUSINESS || PLAN_ECOMMERCE ? 'business' : 'personal';
```

Comment:
We need to probably handle monthly and Jetpack plans here as well. I wonder if we can include a new util in `lib/plans/constants` to centralize the logic:

```
export const PLANS_FOR_BUSINESS_CUSTOMER_TYPE = [
  PLAN_BUSINESS_MONTHLY,
  PLAN_BUSINESS,
  PLAN_BUSINESS_2_YEARS,
  PLAN_ECOMMERCE,
  PLAN_ECOMMERCE_2_YEARS,
  PLAN_JETPACK_BUSINESS,
  PLAN_JETPACK_BUSINESS_MONTHLY,
];

export function isPlanForBusinessCustomerType( plan ) {
	return PLANS_FOR_BUSINESS_CUSTOMER_TYPE.includes( plan );
}
```

Reply:
Actually, just found out that we have [`chooseDefaultCustomerType`](https://github.com/Automattic/wp-calypso/blob/7ace25447946a9188ec4cb667b29b5314f4ed02a/client/lib/plans/index.js#L466), where we get the customer type in `client/my-sites/plans-features-main/index.jsx`:

https://github.com/Automattic/wp-calypso/blob/0f3e445a83f2c54c933564a37d78caa92f014d9c/client/my-sites/plans-features-main/index.jsx#L554-L558

I wonder if we actually have a bug there so it returns the wrong customer type for the given plan.

Reply:
I think there's two issues which arise from trying to change Jetpack plans:

- Whereas currently, the user could still see all plans after toggling, making this change with Jetpack plans could cause an issue where people are confused why the Personal and Premium plans don't appear when there's no immediate 
- It looks like Jetpack nudges record the feature rather than the plan, unlike WordPress.com ones? 

> I wonder if we actually have a bug there so it returns the wrong customer type for the given plan.

If changing that, should the `business` customerType always exclude the Premium plan? That feels like a bigger and wider change rather than a small bug fix, so this PR just proposes excluding it immediately after a nudge has been clicked for now, but happy to explore that.

Reply:
Let me analyze a bit further the code, but my assumption is that upsells already link to the plans comparison page, passing the needed plan for the feature/plan highlighted in the upsell as a `?plan=` query argument. 

They shouldn't be aware of the customer type. To me, that's a responsibility for `chooseDefaultCustomerType`. I think that changes on this PR could hide the underlying issue, since they are overriding/ignoring whatever `chooseDefaultCustomerType` returns.

Reply:
Alright, I think the problem is not in `chooseDefaultCustomerType` at all. For some reason, we want to show the Premium plan to both customer types, so let's leave that as-is. 

My main concern is that we're adding logic to the upsells that IMO should be handled entirely in the plans comparison, in case we link to that page from other places.

@Aurorum let me know what do you think about this alternate approach. 

Provided that:
- [All upsells link to `/plans/:site/?feature=:feature&plan=:plan`](https://github.com/Automattic/wp-calypso/blob/182ec54971af1e1dbac2bc3670064391e30fc79b/client/blocks/upsell-nudge/index.jsx#L60).
- The passed plan is available as the [`selectedPlan` property](https://github.com/Automattic/wp-calypso/blob/182ec54971af1e1dbac2bc3670064391e30fc79b/client/my-sites/plans-features-main/index.jsx#L512) in `PlansFeaturesMain`.
- Our goal is to hide plans not containing the upsold feature.

What if we just hide any plan that is "lower" than the selected plan?

```
const visiblePlans = this.getVisiblePlansForPlanFeatures( plans );
const availablePlans = visiblePlans.filter( plan => {
  if ( isEcommercePlan( this.props.selectedPlan ) ) {
    return isEcommercePlan( plan );
  }
  if ( isBusinessPlan( this.props.selectedPlan ) ) {
    return isBusinessPlan( plan ) || isEcommercePlan( plan );
  }
  if ( isPremiumPlan( this.props.selectedPlan ) ) {
    return isPremiumPlan( plan ) || isBusinessPlan( plan ) || isEcommercePlan( plan );
  }
  // ...
} );
```

Reply:
Thanks @mmtr, that works really well! I've also added a way to "reset" the plans through the "View all plans" link, as I think that's necessary with this approach to prevent confusion. 

<img width="1177" alt="Screenshot 2020-04-16 at 11 41 56" src="https://user-images.githubusercontent.com/43215253/79447144-53de5f80-7fd7-11ea-8254-ba2b91df4392.png">